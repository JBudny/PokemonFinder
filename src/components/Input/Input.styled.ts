import { TextInput, TextInputProps, View } from 'react-native'

import styled from 'styled-components/native'

export const StyledInputWrapper = styled(View)`
	background-color: ${({ theme }) => theme.palette.secondary};
	min-height: ${({ theme }) => theme.designSystem.minTouchTargetSize}px;
	min-width: ${({ theme }) => theme.designSystem.minTouchTargetSize}px;
	border-radius: ${({ theme }) => theme.designSystem.borderRadius}px;
	flex-direction: row;
	align-items: center;
	padding: ${({ theme }) => theme.designSystem.spacingMedium}px;
`

export const StyledIconWrapper = styled(View)`
	justify-content: center;
	align-items: center;
	margin-right: ${({ theme }) => theme.designSystem.spacingMedium}px;
`

export const StyledTextInput = styled(TextInput).attrs<TextInputProps>(
	({ theme }) => ({
		placeholderTextColor: theme.palette.textLight
	})
)`
	flex: 1;
	font-size: ${({ theme }) => theme.designSystem.bodyTextMediumSize}px;
	color: ${({ theme }) => theme.palette.text};
	padding: 0;
`
