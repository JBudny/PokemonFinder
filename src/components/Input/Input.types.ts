import { TextInputProps } from 'react-native'

import { IconProp } from '@fortawesome/fontawesome-svg-core'

export interface InputProps extends TextInputProps {
	icon?: IconProp
}
