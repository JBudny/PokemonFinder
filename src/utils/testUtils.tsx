import React from 'react'

import type { PreloadedState } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react-native'
import { render } from '@testing-library/react-native'

import type { AppStore, RootState } from '@src/store'
import { setupStore } from '@src/store'

import { Providers } from './Providers'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store. For
// future dependencies, such as wanting to test with react-router, you can extend
// this interface to accept a path and route and use those in a <MemoryRouter />
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>
	store?: AppStore
}

const renderWithProviders = (
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {}
) => {
	const Wrapper: React.FC = ({ children }) => (
		<Providers store={store}>{children}</Providers>
	)

	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export { renderWithProviders }
