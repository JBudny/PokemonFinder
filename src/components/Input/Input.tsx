import React, { FC } from 'react'

import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { useTheme } from 'styled-components/native'

import {
	StyledIconWrapper,
	StyledInputWrapper,
	StyledTextInput
} from './Input.styled'
import { InputProps } from './Input.types'

/**
 * Gets required parameters and renders a text input.
 * @param icon
 * EXTENDS: TextInput
 */
export const Input: FC<InputProps> = ({ icon, ...props }) => {
	const { palette } = useTheme()

	return (
		<StyledInputWrapper>
			{icon && (
				<StyledIconWrapper testID="icon">
					<FontAwesomeIcon size={20} color={palette.textLight} icon={icon} />
				</StyledIconWrapper>
			)}
			<StyledTextInput {...props} />
		</StyledInputWrapper>
	)
}
