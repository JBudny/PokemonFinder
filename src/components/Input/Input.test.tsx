import React from 'react'

import { faSearch } from '@fortawesome/free-solid-svg-icons'

import { renderWithProviders } from '@utils/testUtils'

import { Input } from './Input'

test('Input component should display icon when icon prop is passed', () => {
	const { getByTestId } = renderWithProviders(<Input icon={faSearch} />)

	expect(getByTestId('icon')).toBeDefined()
})

test('Input component should not display icon when icon prop is not passed', () => {
	const { queryByTestId } = renderWithProviders(<Input />)

	expect(queryByTestId('icon')).toBeNull()
})

test('Input component should match snapshot', () => {
	const input = renderWithProviders(<Input icon={faSearch} />)

	expect(input).toMatchSnapshot()
})
