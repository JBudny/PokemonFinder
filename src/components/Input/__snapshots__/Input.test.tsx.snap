// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "borderRadius": 15,
        "flexDirection": "row",
        "minHeight": 48,
        "minWidth": 48,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginRight": 10,
        },
      ]
    }
    testID="icon"
  >
    <
      color="rgb(150, 150, 150)"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              128269,
              "search",
            ],
            "f002",
            "M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={20}
    />
  </View>
  <TextInput
    placeholderTextColor="rgb(150, 150, 150)"
    style={
      Array [
        Object {
          "color": "rgb(33, 26, 29)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 20,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  />
</View>
`;
